text/0005-architecture-reviews.md

```diff
@@ -0,0 +1,135 @@
+# Architecture Review Process
+
+## Introduction
+
+Architectural Review decisions within the Firefox organization fall into two categories: "Roadmap" and “Design”.
+
+Product Reviews are not the scope of this process. This process is about architectural change, which should ultimately be in support of product features, but will generally have an effect on multiple product features. Architectural change can be growing new skills or in reducing large scale technical debt.
+
+The function of a **Roadmap Review** is to decide if a thing should be done. The goal is to bring together a packet of data to inform a management decision to provide resources to make the thing happen. A Roadmap Review should happen early in the process so that build time isn’t wasted on a "No" decision, but so that enough information is available to management.
```

Comment:
I think it's fair to say it's more holistic (which is both good and bad): it tries (IIRC) to both consider whether there's a problem, and also consider possible outcomes.

The reason I mention here the possibility of having more than a yes/no outcome is that a yes/no outcome treats the RR as an oracle, and I think it's likely that it will naturally end up in a more productive place: as a discussion that's seeded with a coherent set of observations, and ends up with an agreed description of a problem and a possible roadmap.

A draft roadmap might also be _input_, but I see no reason why a negotiated roadmap might not also be _output_.

Do you agree with my (non-exclusive) list of potential outcomes? Am I reading the process correctly?